workflow-runtime/src/test/java/com/squareup/workflow1/WorkflowInterceptorTest.kt

```diff
@@ -73,7 +82,173 @@ class WorkflowInterceptorTest {
+
+    val fakeContext = object : BaseRenderContext<String, String, String> {
+      override val actionSink: Sink<WorkflowAction<String, String, String>> =
+        Sink { value -> actions += value }
+
+      override fun <ChildPropsT, ChildOutputT, ChildRenderingT> renderChild(
+        child: Workflow<ChildPropsT, ChildOutputT, ChildRenderingT>,
+        props: ChildPropsT,
+        key: String,
+        handler: (ChildOutputT) -> WorkflowAction<String, String, String>
+      ): ChildRenderingT = fail()
+
+      override fun runningSideEffect(
+        key: String,
+        sideEffect: suspend CoroutineScope.() -> Unit
+      ) = fail()
+    }
+
+    val rendering =
+      intercepted.render("props", "string", RenderContext(fakeContext, TestActionWorkflow))
+
+    assertTrue(actions.isEmpty())
+    rendering.onEvent()
+    assertTrue(actions.size == 1)
+
+    assertEquals(
+      listOf("BEGIN|onRender", "END|onRender", "BEGIN|onActionSent", "END|onActionSent"),
+      recorder.consumeEventNames()
+    )
+  }
+
+  @Test fun `intercept() intercepts side effects`() {
+    val recorder = RecordingWorkflowInterceptor()
+    val workflow = TestSideEffectWorkflow()
+    val intercepted = recorder.intercept(workflow, workflow.session)
+    val fakeContext = object : BaseRenderContext<String, String, String> {
+      override val actionSink: Sink<WorkflowAction<String, String, String>> get() = fail()
+
+      override fun <ChildPropsT, ChildOutputT, ChildRenderingT> renderChild(
+        child: Workflow<ChildPropsT, ChildOutputT, ChildRenderingT>,
+        props: ChildPropsT,
+        key: String,
+        handler: (ChildOutputT) -> WorkflowAction<String, String, String>
+      ): ChildRenderingT = fail()
+
+      override fun runningSideEffect(
+        key: String,
+        sideEffect: suspend CoroutineScope.() -> Unit
+      ) {
+        runBlocking { sideEffect() }
+      }
+    }
+
+    intercepted.render("props", "string", RenderContext(fakeContext, workflow))
+
+    assertEquals(
+      listOf(
+        "BEGIN|onRender",
+        "BEGIN|onSideEffectRunning",
+        "END|onSideEffectRunning",
+        "END|onRender"
+      ),
+      recorder.consumeEventNames()
+    )
+  }
+
+  @Test fun `intercept() uses interceptor's context for side effect`() {
+    val recorder = object : RecordingWorkflowInterceptor() {
+      override fun <P, S, O, R> onRender(
+        renderProps: P,
+        renderState: S,
+        proceed: (P, S, RenderContextInterceptor<P, S, O>?) -> R,
+        session: WorkflowSession
+      ): R {
+        return proceed(renderProps, renderState, object : RenderContextInterceptor<P, S, O> {
+          override suspend fun onSideEffectRunning(
+            key: String,
+            proceed: suspend () -> Unit
+          ) {
+            val context = coroutineContext + coroutineContext.job + CoroutineName("intercepted!")
+            CoroutineScope(context).run {
+              proceed()
+            }
```

Comment:
This is a no-op, since `proceed` doesn't take the `CoroutineScope` as a receiver. You want this:
```suggestion
            withContext(CoroutineName("intercepted!")) {
              proceed()
            }
```

Reply:
One might ask, why not just make `proceed` take the scope as a receiver? Because it still needs to be a suspend function, and that would make it possible for the interceptor to provide a different coroutine context implicitly vs via the scope receiver, which is invalid. By forcing the implicit context as the only way to change the context to `proceed`, we eliminate that potential ambiguity.